spec/features/teacher_assigns_resource_spec.rb

```diff
@@ -0,0 +1,134 @@
+      visit "/browse/eresources/1"
+      click_on "Assign"
+      using_wait_time(1) do
+        within("#assign-col") do
+          click_on("Register")
+        end
+        expect(page).to have_css("div#signup-default-modal")
+      end
+    end
+
+    scenario "Anonymous user can click login button in assign dialog modal.", :js => true do
+      visit "/browse/eresources/1"
+      click_on "Assign"
+      using_wait_time(1) do
+        within("#assign-col") do
+          click_on("Log In")
+        end
+        expect(page).to have_css("div#login-default-modal")
+      end
+    end
+
+    scenario "Anonymous user can click copy button in assign dialog modal.", :js => true do
+      visit "/browse/eresources/1"
+      click_on "Assign"
+      using_wait_time(1) do
+        within("#share-col") do
+          click_on("Copy")
+          expect(page).to have_content("Copied to clipboard!")
+        end
+      end
+    end
+
+  end
+
+  context "when logged in as a teacher who has not set up classes yet." do
+    let!(:teacher_user)            { FactoryBot.create(:confirmed_user,
+                                                       :login => "teacher_user",
+                                                       :password => "password",
+                                                       :first_name => 'George',
+                                                       :last_name => 'Christopher') }
+    let!(:teacher_without_classes) { FactoryBot.create(:portal_teacher,
+                                                       :user => teacher_user) }
+    before do
+      login_with_ui_as('teacher_user', 'password')
+    end
+
+    scenario 'Teacher user should see assign button', :js => true do
+      visit "/browse/eresources/1"
+      expect(page).to have_content("Assign")
+    end
+
+    scenario 'Teacher user should be able to click assign button and see assign dialog modal, but will not see a list of classes.', :js => true do
+      # first, archive sample class
+      visit "/portal/classes/manage"
+      first("input[type='checkbox']").click
+      visit "/browse/eresources/1"
+      click_on "Assign"
+      using_wait_time(1) do
+        expect(page).to_not have_content("Register")
+        expect(page).to_not have_content("Log In")
+        expect(page).to_not have_content("Select the class(es) you want to assign this resource to below.")
+        expect(page).to have_content("You don't have any active classes.")
+      end
+    end
+
+    scenario "Teacher user can click copy button in assign dialog modal.", :js => true do
+      visit "/browse/eresources/1"
+      click_on "Assign"
+      using_wait_time(1) do
+        within("#share-col") do
+          click_on("Copy")
+          expect(page).to have_content("Copied to clipboard!")
+        end
+      end
+    end
+  end
+
+  context "when logged in as a teacher who has set up classes already." do
```

Comment:
I suggest adding more more context at the top level: "when logged in as a student"